pkg/query-service/app/integrations/messagingQueues/kafka/sql.go

```diff
@@ -0,0 +1,171 @@
+package kafka
+
+import (
+	"fmt"
+)
+
+func generateConsumerSQL(start, end int64, topic, partition, consumerGroup, queueType string) string {
+	query := fmt.Sprintf(`
+WITH 
+-- Sub query for p99 calculation
+p99_query AS (
+    SELECT
+        serviceName,
+        quantile(0.99)(durationNano) / 1000000 as p99
+    FROM signoz_traces.signoz_index_v2
+    WHERE
+        timestamp >= '%d'
+      AND timestamp <= '%d'
+      AND kind = 5
+      AND msgSystem = '%s' 
+      AND stringTagMap['messaging.destination.name'] = '%s'
+      AND stringTagMap['messaging.destination.partition.id'] = '%s'
+	  AND stringTagMap['messaging.kafka.consumer.group'] = '%s'
+    GROUP BY serviceName
+),
+
+-- Sub query for RPS calculation
+rps_query AS (
+    SELECT
+        serviceName,
+        count(*) / ((%d - %d) / 1000000000) AS rps  -- Convert nanoseconds to seconds
```

Comment:
I hope you made sure there are no overflow issue with `(%d - %d)` part

Reply:
I'm not handling the cases in query, but given the time ranges will be short for the Messaging queue, handling them as `errored response` would be enough.
Though, a check can be added before running the query (in the code itself, not in the query), WDYT? Should I add a check?
If yes, then the max limit is not known (the user would usually search for a week's data, I can add a check for a negative value, but I don't think UI would be doing that (logically from the time selector). So not sure what should be a reasonable range check here, before running the query...

Reply:
can you maybe pass the denominator directly?

Reply:
done